From 52c86f01133e24a86294e51c9d5eb1b9eb805eda Mon Sep 17 00:00:00 2001
From: Mark Brown <broonie@linaro.org>
Date: Tue, 25 Mar 2014 19:28:22 +0000
Subject: [PATCH] spi: Do not require a completion

There is no real reason why we require transfers to have a completion and
the only user of the completion now checks to see if one has been provided
before using it so stop enforcing this. This makes it more convenient for
drivers to chain multiple asynchronous transfers together.

Signed-off-by: Mark Brown <broonie@linaro.org>
(cherry picked from commit 1e25cd4729bd76662d02b142a6a7f8504bb6aea7)
(cherry picked from commit 3d780e874e52daf0a976460a62f520cb438b6743)
---
 drivers/spi/spi.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index 30dfa64..568cde3 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -1827,8 +1827,6 @@ static int __spi_validate(struct spi_device *spi, struct spi_message *message)
 
 	if (list_empty(&message->transfers))
 		return -EINVAL;
-	if (!message->complete)
-		return -EINVAL;
 
 	/* Half-duplex links include original MicroWire, and ones with
 	 * only one data pin like SPI_3WIRE (switches direction) or where
-- 
1.8.0

